dartagnan/src/main/java/com/dat3m/dartagnan/program/memory/Memory.java

```diff
@@ -16,16 +17,16 @@ public class Memory {
      * @return
      * Points to the created location.
      */
-    public MemoryObject allocate(int size, boolean isStatic) {
+    public MemoryObject allocate(IntegerType addressType, int size, boolean isStatic) {
         Preconditions.checkArgument(size > 0, "Illegal malloc. Size must be positive");
-        MemoryObject address = new MemoryObject(nextIndex++, size, isStatic);
+        MemoryObject address = new MemoryObject(nextIndex++, addressType, size, isStatic);
```

Comment:
Is this necessary? Since a memory belongs to a single program, we might as well have `Memory` contain a reference to `Program` from which it can obtain the necessary types.